File: src/main/java/io/reactivesocket/cli/TimeUtil.java
package io.rsocket.cli;

import io.airlift.airline.ParseException;

import java.time.Duration;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class TimeUtil {
    private static final Pattern DURATION_FORMAT = Pattern.compile("(\\d+)(ms|s|m)");

    public static Duration parseShortDuration(String keepalive) {
        Matcher match = DURATION_FORMAT.matcher(keepalive);

        if (!match.matches()) {
            throw new ParseException("Unknown duration format '" + keepalive + "'");
        }

        long amount = Long.valueOf(match.group(1));
        String unit = match.group(2);

        switch (unit) {
            case "ms":
                return Duration.ofMillis(amount);
            case "s":
                return Duration.ofSeconds(amount);
            default:
                return Duration.ofMinutes(amount);
        }
    }
}

File: src/main/kotlin/io/rsocket/cli/util/TimeUtil.kt
package io.rsocket.cli.util

import io.airlift.airline.ParseException
import java.time.Duration
import java.util.regex.Pattern

object TimeUtil {
    private val DURATION_FORMAT = Pattern.compile("(\\d+)(ms|s|m)")

    fun parseShortDuration(keepalive: String): Duration {
        val match = DURATION_FORMAT.matcher(keepalive)

        if (!match.matches()) {
            throw ParseException("Unknown duration format '$keepalive'")
        }

        val amount = java.lang.Long.valueOf(match.group(1))!!
        val unit = match.group(2)

        return when (unit) {
            "ms" -> Duration.ofMillis(amount)
            "s" -> Duration.ofSeconds(amount)
            else -> Duration.ofMinutes(amount)
        }
    }
}
